webapp/publisher/snaps/views.py

```diff
@@ -1308,3 +1309,53 @@ def post_preview(snap_name):
     context["normalized_os"] = preview_data.get_normalised_oses()
 
     return flask.render_template("store/snap-details.html", **context)
+
+
+@publisher_snaps.route("/<snap_name>/builds", methods=["GET"])
+@login_required
+def get_snap_builds(snap_name):
+    try:
+        details = api.get_snap_info(snap_name, flask.session)
+    except ApiResponseErrorList as api_response_error_list:
+        if api_response_error_list.status_code == 404:
+            return flask.abort(404, "No snap named {}".format(snap_name))
+        else:
+            return _handle_error_list(api_response_error_list.errors)
+    except ApiError as api_error:
+        return _handle_errors(api_error)
+
+    context = {
+        "snap_id": details["snap_id"],
+        "snap_name": details["snap_name"],
+        "snap_title": details["title"],
+    }
+
+    snap = launchpad.snaps.findByStoreName(
+        owner="https://api.launchpad.net/devel/~build.snapcraft.io",
+        store_name=details["snap_name"],
+    )
+
+    builds = []
+    if len(snap) >= 1:
+        for build in snap[0].builds:
+            builds.append(
+                {
+                    "arch_tag": build.arch_tag,
+                    "channels": build.channels,
+                    "datebuilt": build.datebuilt,
+                    "duration": build.duration,
+                    "link": build.web_link,
+                    "revision_id": build.revision_id,
+                    "title": build.title,
+                }
```

Comment:
I'm not sure what `channels` are in terms of builds, I don't know if it has anything to do with store channels, so we may not need it.

What we want for sure is build status.
If I recall correctly there are 2 fields - status of the build itself, and status of the store upload, and in BSI we combine them into one somehow - I don't recall the exact logic. Maybe we could replicate it here, or at least have both fields in the context for front-end to figure out the status.

For the `web_link` - I'm not sure if we want to link directly to LP build page. It may actually be useful to link to BSI page instead. This would require building the URL from BSI domain + github handle + build id. We can do it on front-end, but it's a good candidate for backend logic as well.

Please also include the log link.